lib/openssl/ccm.rb

```diff
@@ -45,7 +46,8 @@ def initialize(cipher, key, mac_len)
         fail CCMError, 'invalid mac length'
       end
 
-      @cipher = OpenSSL::Cipher.new("#{cipher}-128-CBC")
+      @cipher = OpenSSL::Cipher.new("#{cipher}-" + (key.length * 8).to_s +
```

Comment:
What would happen when somebody use a key with length of 17 byte?
Before your changes it doesnt matter because the cipher ignores the additional byte.
But when u want to choose the Cipher depending on the key length it needes to be:
128 for key.length < 24
192 for key.length < 32
256 for else